pkg/aggregator/mocksender/asserts_test.go

```diff
@@ -18,35 +18,49 @@ type unittestMock struct {
 	mock.Mock
 }
 
-// Test the behavior of AnythingBut
-// In this method we could observe an intend failure of "AssertNotCalled" tied to the localTester
-// Method returns the failure state of the localTester
-func (u *unittestMock) assertAnythingBut(t *testing.T, s string) bool {
-	u.Mock.AssertCalled(t, "dummyMethod", mock.AnythingOfType("string"))
-
-	// Create a local testing.T just for the following AssertNotCalled
-	localTester := &testing.T{}
-	// Pass the [localTester *testing.T] instead of [t *testing.T]
-	u.Mock.AssertNotCalled(localTester, "dummyMethod", AnythingBut(s))
-	return localTester.Failed()
-}
-
 // A dummy Method to be called
-func (u *unittestMock) dummyMethod(s string) {
+func (u *unittestMock) dummyMethod(s interface{}) {
 	u.Called(s)
 }
 
+// In this method we could observe an intend failure of "AssertNotCalled" tied to the localTester
 func TestAnythingBut(t *testing.T) {
 	m := &unittestMock{}
 	m.On("dummyMethod", mock.AnythingOfType("string")).Return()
-	m.dummyMethod("ok")
 
-	result := m.assertAnythingBut(t, "ok")
-	assert.False(t, result)
+	m.dummyMethod("ok")
+	m.AssertCalled(t, "dummyMethod", mock.AnythingOfType("string"))
 
 	m.dummyMethod("ko")
-	result = m.assertAnythingBut(t, "ok")
-	assert.True(t, result)
+	m.AssertCalled(t, "dummyMethod", mock.AnythingOfType("string"))
+
+	// Create a local testing.T just for the following AssertNotCalled
+	localTester := &testing.T{}
+	// Pass the [localTester *testing.T] instead of [t *testing.T]
+	m.AssertNotCalled(localTester, "dummyMethod", AnythingBut("ok"))
+	// Expected a failure on localTester
+	assert.True(t, localTester.Failed())
+}
+
+// In this method we could observe an intend failure of "AssertNotCalled" tied to the localTester
+func TestIsGreaterOrEqual(t *testing.T) {
+	m := &unittestMock{}
+	m.On("dummyMethod", mock.AnythingOfType("float64")).Return()
+
+	const dummyValue = 3.0
+	m.dummyMethod(dummyValue)
+
+	for i := 0.0; i < dummyValue+1; i++ {
+		m.AssertCalled(t, "dummyMethod", IsGreaterOrEqual(i))
+	}
+	m.AssertNotCalled(t, "dummyMethod", IsGreaterOrEqual(dummyValue+1))
+
+	// Create a local testing.T just for the following AssertCalled
```

Comment:
Not sure if it's relevant to check the fail status on `localTester testing.T`

Reply:
Let's keep it, to test for false positives